control_software/src/paper_gpu/file_conversion.py

```diff
@@ -0,0 +1,349 @@
+        frequency channels in data; unsigned integer), nstokes (number of
+        stokes/polarizations; unsigned integer), corr_ver (correlator git
+        version; string), tag (observation tag; string), ant_0_array (list of
+        first antennas in data; 1-d array of ints of length nblts), ant_1_array
+        (list of second antennas in data; 1-d array of ints of length nblts),
+        time_array (JD of observation; 1-d array of floats of length nblts),
+        integration_time (seconds of observation; 1-d array of floats of length
+        nblts).
+    """
+    # pull data from HDF5 metadata file
+    with h5py.File(filename, "r") as h5f:
+        t0 = h5f["t0"][()]
+        mcnt = h5f["mcnt"][()]
+        nfreq = h5f["nfreq"][()]
+        nstokes = h5f["nstokes"][()]
+        corr_ver = h5f["corr_ver"][()].decode("utf-8")
+        tag = h5f["tag"][()].decode("utf-8")
+        ant_0_array = h5f["ant_0_array"][()]
+        ant_1_array = h5f["ant_1_array"][()]
+        time_array = h5f["time_array"][()]
+        integration_time = h5f["integration_time"][()]
+
+    return (
+        t0,
+        mcnt,
+        nfreq,
+        nstokes,
+        corr_ver,
+        tag,
+        ant_0_array,
+        ant_1_array,
+        time_array,
+        integration_time,
+    )
+
+
+def read_data_file(filename, data_shape):
+    """
+    Read a block of binary data from file.
+
+    Parameters
+    ----------
+    filename : str
+        The name of the file to read.
+    data_shape : tuple of int
+        The expected size of the data. Data read will be reshaped to this.
+
+    Returns
+    -------
+    data : nd-array
+        The data that was read. Compound numpy datatype with a "r" field and "i"
+        field, both 32-bit integers.
+
+    Raises
+    ------
+    ValueError
+        Raised if the data read in cannot be reshaped into the specified shape.
+    """
+    # read raw binary data
+    data = np.fromfile(filename, dtype=_hera_corr_dtype)
+    try:
+        data = data.reshape(data_shape)
+    except ValueError:
+        raise ValueError(
+            f"data cannot be reshaped; data read is {data.size} elements, "
+            "target size is {data_shape}"
+        )
+
+    # return to user
+    return data
+
+
+def get_antpos_info():
+    """
+    Fetch HERA antenna positions from hera_mc.
+
+    Parameters
+    ----------
+    None
+
+    Returns
+    -------
+    antpos_xyz : ndarray
+        An array of floats of size (350, 3) which contains the antenna
+        positions in XYZ (i.e., ECEF) coordinates.
+    ant_names : ndarray
+        An array of strings of size (350,) which contains the antenna names.
+    """
+    # read antenna positions from M&C
+    ants = geo_sysdef.read_antennas()
+    # convert from eastings/northings to ENU
+    # HERA is in Zone 34J; corresponds to latitude 10000000 in northings
+    latlon_p = ccrs.Geodetic()
+    utm_p = ccrs.UTM(34)
```

Comment:
Magic numbers that could live in constants or default parameters

Reply:
Done in 25c0e2f.